### PR TITLE
PHOENIX-6969 Projection bug in hinted uncovered index query with order by

### PR DESCRIPTION
**Pull request by stoty**

### CODE REVIEW

File: phoenix-core/src/test/java/org/apache/phoenix/compile/QueryCompilerTest.java [hunk 1]
```diff
@@ -6981,4 +6981,31 @@ public void testReverseVarLengthRange6916() throws Exception {
             assertEquals(explainExpected, explainPlan);
         }
     }
+
+    @Test
+    public void testUncoveredPhoenix6969() throws Exception {
+
+        try (Connection conn = DriverManager.getConnection(getUrl());
+                Statement stmt = conn.createStatement();) {
```
Comment (virajjasani) [inline phoenix-core/src/test/java/org/apache/phoenix/compile/QueryCompilerTest.java:6989]: nit: extra `;`?

File: phoenix-core/src/test/java/org/apache/phoenix/compile/QueryCompilerTest.java [hunk 2]
```diff
@@ -6981,4 +6981,31 @@ public void testReverseVarLengthRange6916() throws Exception {
...
+
+            stmt.execute(
+                "create table dd (k1 integer not null, k2 integer not null, k3 integer not null,"
```
Comment (virajjasani) [inline phoenix-core/src/test/java/org/apache/phoenix/compile/QueryCompilerTest.java:6992]: table name to be `generateUniqueName()`?

File: phoenix-core/src/test/java/org/apache/phoenix/compile/QueryCompilerTest.java [hunk 3]
```diff
@@ -6981,4 +6981,31 @@ public void testReverseVarLengthRange6916() throws Exception {
...
+                + " k4 integer not null, v1 integer, v2 integer, v3 integer, v4 integer"
+                + " constraint pk primary key (k1,k2,k3,k4))");
+            stmt.execute("create index ii on dd (k4, k1, k2, k3)");
```
Comment (virajjasani) [inline phoenix-core/src/test/java/org/apache/phoenix/compile/QueryCompilerTest.java:6995]: same for index name

File: phoenix-core/src/main/java/org/apache/phoenix/schema/IndexDataColumnRef.java [hunk 4]
```diff
@@ -36,6 +35,7 @@
  */
 public class IndexDataColumnRef extends ColumnRef {
     final private int position;
+    // Despite the final keyword, columns IS mutable, and must not be used for equality/HashMap
```
Comment (virajjasani) [inline phoenix-core/src/main/java/org/apache/phoenix/schema/IndexDataColumnRef.java:38]: nit: `equality/hashcode`

**Review by virajjasani — APPROVED** [review]
Left few nits, +1 otherwise

File: phoenix-core/src/test/java/org/apache/phoenix/compile/QueryCompilerTest.java (hunk 1 above)
Comment (stoty) [reply phoenix-core/src/test/java/org/apache/phoenix/compile/QueryCompilerTest.java:6989]: Thx, done

File: phoenix-core/src/test/java/org/apache/phoenix/compile/QueryCompilerTest.java (hunk 2 above)
Comment (stoty) [reply phoenix-core/src/test/java/org/apache/phoenix/compile/QueryCompilerTest.java:6992]: Not needed for these unit tests, there is no state preserved between connections.

File: phoenix-core/src/test/java/org/apache/phoenix/compile/QueryCompilerTest.java (hunk 3 above)
Comment (stoty) [reply phoenix-core/src/test/java/org/apache/phoenix/compile/QueryCompilerTest.java:6995]: Not needed for these unit tests, there is no state preserved between connections.

File: phoenix-core/src/main/java/org/apache/phoenix/schema/IndexDataColumnRef.java (hunk 4 above)
Comment (stoty) [reply phoenix-core/src/main/java/org/apache/phoenix/schema/IndexDataColumnRef.java:38]: Thx, fixed.

**Pull request closed without merging**